mongo/dataset_store.go

```diff
@@ -607,3 +615,41 @@ func (m *Mongo) CheckEditionExists(id, editionID, state string) error {
 
 	return nil
 }
+
+func (m *Mongo) Ping(ctx context.Context) (time.Time, error) {
+	if time.Since(m.lastPingTime) < 1*time.Second {
+		return m.lastPingTime, m.lastPingResult
+	}
+
+	s := m.Session.Copy()
+	defer s.Close()
+
+	m.lastPingTime = time.Now()
+	pingDoneChan := make(chan error)
+	var wg sync.WaitGroup
+
+	wg.Add(1)
+	go func() {
+		log.Trace("db ping", nil)
+		var res interface{}
+		var err error
+		if err = s.DB(m.Database).C(editionsCollection).Find(nil).One(&res); err != nil {
```

Comment:
This could be replaced with https://godoc.org/gopkg.in/mgo.v2#Session.Ping

Reply:
good idea, will do that